main.py

```diff
@@ -1,20 +1,27 @@
-from db_connection import build_full_url, make_engine, get_session
-from db_engines.sql_server_engine import url_SQLServerTestDBMS
-from db_engines.db_sources_data.sql_server_test_localhost import dbdriver, dbpath, username, password, SQLServerTestDBs
+import json
+from datetime import datetime
+
+import pandas as pd
+from pandas.plotting._matplotlib.style import get_standard_colors
+# Get default color style from pandas - can be changed to any other color list
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+from data_analysis.defect_event_root_cause import join_defect_event_root_cause_filter_behaviour_id, BehaviourPattern
+from phillip.db_connection import get_session
+from db_engines.sql_server_engine import engine_sqlservertest_main as sqlserver_engine
 
 from loguru import logger
 from loguru_logging.debug_formatter import debug_format
 
+import dql_scripts
 import dql_scripts.simple_select as smpl_sel
 import dql_scripts.select_joins as jnt_sel
-from ddl_scripts.creating_tables import SignalMeta
-from ddl_scripts.adding_data import add_new_objs, test_dataobj1, test_dataobj2
 
-debug_format()
-url = build_full_url(url_SQLServerTestDBMS + SQLServerTestDBs.MASTER_DB.value + "?driver=ODBC+Driver+17+for+SQL+Server")
-global_engine = make_engine(url, 1000)
-global_session = get_session(global_engine)
+debug_format()  # adds a custom format for debug-level of loguru (saving local .log-files)
+global_session = get_session(sqlserver_engine)
```

Comment:
Ich glaube diese plotting Sachen würde ich in ein extra modul packen, also nicht in main.py.

Hast du das mal mit Testdaten ausprobiert ob das läuft ?

Reply:
Yep, ich habe nun ein Skript plots.py für die Matplotlib-basierten Methoden erstellt. Die Methode fürs Gruppieren der Signaldaten von Events habe ich ins dataframes.py Skript verschoben. 

Es sind nun auch DocString zu jeder Methode drin sowie ein Paar To-Dos 😃 

Leider kann ich keine Screenshots mehr auf dem Master Client auf GutHub hochladen... 🙄 Nach dem ich die Datanbank restrukturiert habe, habe ich tatsächlich mit ein Paar klinDensätzten signal_data aus der KundenDB bezogen und Plots erstellt! Es klappt alles gut 🏅